sources/file_table.cpp

```diff
@@ -175,6 +177,15 @@ FileTable::~FileTable()
 FileBase* FileTable::open_as(const id_type& id, int type)
 {
     auto it = m_files.find(id);
+    if(it == m_files.end()) {
+        std::lock_guard<std::mutex> l(m_closing_lock);
+        auto close_it = m_files_to_close.find(id);
+        if(close_it != m_files_to_close.end()) {
+            m_files.emplace(id,std::move(close_it->second));
+            m_files_to_close.erase(close_it);
+            it = m_files.find(id);
```

Comment:
`emplace` returns the iterator to the created element. Another `find` is unnecessary time cost.